src/crimson/CMakeLists.txt

```diff
@@ -127,30 +127,17 @@ target_compile_definitions(crimson-common PRIVATE
   "CEPH_INSTALL_FULL_PKGLIBDIR=\"${CEPH_INSTALL_FULL_PKGLIBDIR}\""
   "CEPH_INSTALL_DATADIR=\"${CEPH_INSTALL_DATADIR}\"")
 
-set(crimson_common_deps
-  Boost::iostreams
-  Boost::random
-  json_spirit)
-
-set(crimson_common_public_deps crimson::cflags)
-if(WITH_JAEGER)
-  list(APPEND crimson_common_public_deps jaeger_base)
-endif()
-
-if(WITH_BREAKPAD)
-  list(APPEND crimson_common_deps Breakpad::client)
-endif()
-
-if(NOT WITH_SYSTEM_BOOST)
-  list(APPEND crimson_common_deps ${ZLIB_LIBRARIES})
-endif()
-
 target_link_libraries(crimson-common
   PUBLIC
-    ${crimson_common_public_deps}
+    $<$<BOOL:${WITH_JAEGER}>:jaeger_base>
```

Comment:
nit, the gist of this change is not to use `target_link_libraries()`  -- we are already using it. the thing is that we are replacing condition statements with more compacted generator expressions here.